gensim/api/__init__.py

```diff
@@ -0,0 +1,174 @@
+    filename=log_file_dir, level=logging.INFO)
+console = logging.StreamHandler()
+console.setLevel(logging.INFO)
+logger = logging.getLogger('gensim.api')
+logger.addHandler(console)
+
+
+def download(dataset):
+    url = "https://github.com/chaitaliSaini/Corpus_and_models/releases/download/{f}/{f}.tar.gz".format(f=dataset)
+    data_folder_dir = os.path.join(base_dir, dataset)
+    data = catalogue(print_list=False)
+    corpuses = data['gensim']['corpus']
+    models = data['gensim']['model']
+    if dataset not in corpuses and dataset not in models:
+        logger.error(
+            "Incorect Model/corpus name. Use catalogue(print_list=TRUE) or"
+            " python -m gensim -c to get a list of models/corpuses"
+            " available.")
+        sys.exit(0)
+    compressed_folder_name = "{f}.tar.gz".format(f=dataset)
+    compressed_folder_dir = os.path.join(base_dir, compressed_folder_name)
+    is_installed = False
+    is_downloaded = False
+    installed_message = "{f} installed".format(f=dataset)
+    downloaded_message = "{f} downloaded".format(f=dataset)
+    if os.path.exists(data_folder_dir):
+        log_file_dir = os.path.join(base_dir, 'api.log')
+        with open(log_file_dir) as f:
+            f = f.readlines()
+        for line in f:
+            if installed_message in line:
+                print("{} has already been installed".format(dataset))
+                is_installed = True
+                sys.exit(0)
+    if os.path.exists(data_folder_dir) and not is_installed:
+        for line in f:
+            if downloaded_message in line:
+                is_downloaded = True
+                break
+    if not is_downloaded:
+        if not os.path.exists(data_folder_dir):
+            logger.info("Creating %s", data_folder_dir)
+            os.makedirs(data_folder_dir)
+            if os.path.exists(data_folder_dir):
+                logger.info("Creation of %s successful.", data_folder_dir)
+            else:
+                logger.error(
+                    "Not able to create %s. Make sure you have the correct read/"
+                    "write permissions for %s or you can try creating it manually",
+                    data_folder_dir, base_dir)
+                sys.exit(0)
+        logger.info("Downloading %s", dataset)
+        urllib.urlretrieve(url, compressed_folder_dir)
+        data_url = data_links(dataset)
+        if data_url is not None:
+            index = data_url.rfind("/")
+            data_dir = os.path.join(data_folder_dir, data_url[index + 1:])
+            urllib.urlretrieve(data_url, data_dir)
+        logger.info("%s downloaded", dataset)
+    if not is_installed:
+            tar = tarfile.open(compressed_folder_dir)
+            logger.info("Extracting files from %s", data_folder_dir)
+            tar.extractall(data_folder_dir)
+            tar.close()
+            logger.info("%s installed", dataset)
+
+
+def catalogue(print_list=False):
```

Comment:
I tested it, `print_list` is useless, please remove this arg (and printing code).

Reply:
Also please merge `catalogue` and `info` function to new `info` function (add arguent for concrete model/dataset, by default, it should be None)